web/src/typeahead_helper.ts

```diff
@@ -199,12 +199,15 @@ export let render_stream = (stream: StreamData): string =>
         stream,
     });
 
-export const render_stream_topic = (topic_object: TopicSuggestion): string =>
-    render_typeahead_item({
+export const render_stream_topic = (topic_object: TopicSuggestion): string => {
+    const is_empty_string_topic = topic_object.topic === "";
+    topic_object.topic = util.get_final_topic_display_name(topic_object.topic);
```

Comment:
Should the `TopicSuggestion` constructor instead be doing this? I guess it depends whether we'd want "general ..." to match "general chat"; I think the answer is yes?